### DISCUSSION TITLE
Confusion 1) what is difference between Topic Modelling and Clustering?  2) How CountVectorizor work for tokenization and stop word removal after embedding? is work in numeric data?

### DISCUSSION THREAD

**Discussion by hari-chalise** [MaartenGr/BERTopic#2225] [General]
I am recently work my thesis on Topic Modelling by using BERTopic for nepali text, but i have lots of confusion what exactly difference between topic modelling and clustering. In BERTopic documentation there is step of tokenization after clustering, and my question is that tokenization is perform on numeric data after embedding? please clearify me. And also how exactly topic are formed after C-TF-IDF?

**Comment by MaartenGr** [comment]
> I am recently work my thesis on Topic Modelling by using BERTopic for nepali text, but i have lots of confusion what exactly difference between topic modelling and clustering.

After clustering, you still do not have any descriptions of the clusters right? Here, topic modeling is meant to give descriptions to the clusters that you found as a way to extract topics. It's a bit more involved than that but it is basically the difference between **topic assignment** (clustering) and **topic representation** (topic modeling). 

> In BERTopic documentation there is step of tokenization after clustering, and my question is that tokenization is perform on numeric data after embedding? please clearify me

It performs the tokenization on the aggregated groups of documents, not the embeddings. 

> And also how exactly topic are formed after C-TF-IDF?

It is through c-TF-IDF that it is formed, not after. For more information on the procedure of BERTopic, I would highly advise reading through the documentation [here](https://maartengr.github.io/BERTopic/algorithm/algorithm.html) or the tutorial [here](https://www.maartengrootendorst.com/blog/topicbert/).

**Reply by hari-chalise** [reply]
Thanks a lots @MaartenGr  really appreciated. I look forward if i encountered others queries.

**Comment by hari-chalise** [comment]
Dear @MaartenGr  In BERTopic which similarity measure is applied such as cosine similarity or euclidean distance? Also i have small confusion what is difference between DBSCAN and HDBSCAN instead of basic difference , on the basis of clustering with cluster size, process of clustering. Thanks in Advance.

**Reply by MaartenGr** [reply]
> In BERTopic which similarity measure is applied such as cosine similarity or euclidean distance?

In which step? There are a couple of places where similarity metrics are applied.

> Also i have small confusion what is difference between DBSCAN and HDBSCAN instead of basic difference , on the basis of clustering with cluster size, process of clustering. 

I would advise checking the official HDBSCAN repo, they explain it really well.
